### PR TITLE
Configure LdapUserDao and LdapProfileDao to use TCKs

### PR DESCRIPTION
**Pull request by evoevodin**
Along with tests refactoring i added an ability to use `NamingEnumeration` and `InitialLdapContext` in the _try-with-resources_ statement, which helps to avoid the boilerplate code. 

@skabashnyuk, @garagatyi please review

The merge of this PR depends on #243

### CODE REVIEW

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/ldap/ProfileAttributesMapper.java [hunk 1]
```diff
@@ -46,11 +46,11 @@
 @Singleton
 public class ProfileAttributesMapper {
 
-    private final String              profileDn;
-    private final String              profileIdAttr;
-    private final String              profileContainerDn;
-    private final String              profileEmailAttr;
-    private final Map<String, String> allowedAttributes;
+    final String              profileDn;
```
Comment (garagatyi) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/ldap/ProfileAttributesMapper.java]: If it is needed for test only please add visible for testing annotation

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/ldap/LdapUserDao.java [hunk 2]
```diff
@@ -345,35 +321,12 @@ private String createFilter(String attribute, String value) {
     private Attributes getUserAttributesByFilter(InitialLdapContext context, String filter) throws NamingException {
         final SearchControls controls = new SearchControls();
         controls.setSearchScope(SearchControls.SUBTREE_SCOPE);
-        NamingEnumeration<SearchResult> enumeration = null;
-        try {
-            enumeration = context.search(containerDn, filter, controls);
+        try (CloseableSupplier<NamingEnumeration<SearchResult>> enumSup = deferClose(context.search(containerDn, filter, controls))) {
```
Comment (garagatyi) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/ldap/LdapUserDao.java]: I don't like deferClose method name. As it is verb it says that it does something, but it does not. It just creates wrapper. So method says that context will be closed, but actually without try with resources or similar construction context won't be closed.

**Comment by garagatyi** [comment]
LGTM, but please consider my comments.

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/ldap/LdapUserDao.java (hunk 2 above)
Comment (evoevodin) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/ldap/LdapUserDao.java]: Thank you, i will consider a new name for the method.

**Comment by skabashnyuk** [comment]
ok

**Pull request closed without merging**